mmpy_bot/function.py

```diff
@@ -5,22 +5,27 @@
 import logging
 import re
 from abc import ABC, abstractmethod
-from typing import Callable, Optional, Sequence
+from typing import TYPE_CHECKING, Callable, Optional, Sequence, Union
 
 import click
 
-from mmpy_bot.utils import completed_future, spaces
+from mmpy_bot.utils import completed_future
 from mmpy_bot.webhook_server import NoResponse
 from mmpy_bot.wrappers import Message, WebHookEvent
 
+if TYPE_CHECKING:
+    from mmpy_bot.plugins import Plugin
+
+
 log = logging.getLogger("mmpy.function")
 
 
 class Function(ABC):
     def __init__(
         self,
-        function: Callable,
+        function: Union[Callable, Function],
         matcher: re.Pattern,
+        **annotations,
```

Comment:
Can you rename this `metadata`? `annotations` suggests that these might be the type annotations of the wrapped function 😅

Reply:
Done